### PR TITLE
ci: pull request template

### PR DESCRIPTION
**Pull request by d0choa**
## ✨ Context

As discussed in @RobinM-code in #510, we would benefit from having a PR template so that contributors understand what are the expectations when opening new PRs.

## 🛠 What does this PR implement

A Github PR template. 

## 🙈 Missing

Templates are a very subjective topic. I welcome any feedback you might have cc @RobinM-code @DSuveges @ireneisdoomed 

## 🚦 Checks

- [x] covers one single feature (one change at a time)
- [x] documentation added/updated
- [x] there is no commented out code in this PR
- [x] follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) in PR title and commits.
- [x] is up-to-date with `dev` branch
- [x] tests have been added to verify that the new code works
- [x] passes local tests (`make test`)
- [x] passes pre-commits (e.g `poetry run pre-commit run --all-files`)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+## 🛠 What does this PR implement
+
+<!--- _Detailed description of the changes introduced, Give examples of the changes you've made in this pull request, include an itemized list if you can and
+add diagrams or images if necessary. It'll help the reviewer_ -->
+
+## 🙈 Missing
+
+<!--- If there are things that are requested in the task and were not implemented, list them here -->
+
+## 🚦 Checks
```
Comment (ireneisdoomed) [inline .github/pull_request_template.md:19]:
```suggestion
## 🚦 Before submitting
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+- [ ] covers one single feature (one change at a time)
```
Comment (ireneisdoomed) [inline .github/pull_request_template.md:21]:
```suggestion
- [ ] Do these changes cover one single feature (one change at a time)?
```

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+- [ ] documentation added/updated
```
Comment (ireneisdoomed) [inline .github/pull_request_template.md:22]:
```suggestion
- [ ] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
-  [ ] Did you make sure to update the documentation with your changes?
```

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+- [ ] there is no commented out code in this PR
```
Comment (ireneisdoomed) [inline .github/pull_request_template.md:23]:
```suggestion
- [ ] Did you make sure there is no commented out code in this PR?
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+- [ ] follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) in PR title and commits.
```
Comment (ireneisdoomed) [inline .github/pull_request_template.md:24]:
```suggestion
- [ ] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
```

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+- [ ] is up-to-date with `dev` branch
```
Comment (ireneisdoomed) [inline .github/pull_request_template.md:25]:
```suggestion
- [ ] Did you make sure the branch is up-to-date with the `dev` branch?
```

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+- [ ] tests have been added to verify that the new code works
```
Comment (ireneisdoomed) [inline .github/pull_request_template.md:26]:
```suggestion
- [ ] Did you write any new necessary tests?
```

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+- [ ] passes local tests (`make test`)
```
Comment (ireneisdoomed) [inline .github/pull_request_template.md:27]:
```suggestion
- [ ] Did you make sure the changes pass local tests (`make test`)?
```

File: .github/pull_request_template.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+- [ ] passes pre-commits (e.g `poetry run pre-commit run --all-files`)
```
Comment (ireneisdoomed) [inline .github/pull_request_template.md:28]:
```suggestion
- [ ] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?
```

File: .github/pull_request_template.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
+## ✨ Context
+
+<!--- What's the context for the changes? If the changes are related to a specific issue, please [link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) to it:_
```
Comment (ireneisdoomed) [inline .github/pull_request_template.md:3]:
```suggestion
<!---
Congratulations! You've made it this far!

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your contribution.

What's the context for the changes? If the changes are related to a specific issue, please [link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) to it:_
```

**Review by ireneisdoomed — COMMENTED** [review]
Thank you! I have stylistic suggestions, based on the Transformers PR template
Currently our contribution guidelines are "polluted" with instructions about environment set up, we could improve this.

**Comment by RobinM-code** [comment]
Some nice work @d0choa, this will help for sure!

Also:
> Currently our contribution guidelines are "polluted" with instructions about environment set up, we could improve this.

Agree with this :)

**Comment by DSuveges** [comment]
I think these are fine, if changes will be necessary later, it can always be updated.

**Review by ireneisdoomed — APPROVED** [review]
Ty for the changes!

**Pull request merged**